lib/firebase.rb

```diff
@@ -53,11 +53,18 @@ def update(path, data, query={})
     private
 
     def process(verb, path, data=nil, query={})
+      if non_relative_path_against_child_base_url?(path)
+        raise(ArgumentError.new("Path invalid: #{path}. A relative path must be used against a child base_uri"))
+      end
       Firebase::Response.new @request.request(verb, "#{path}.json", {
         :body             => (data && data.to_json),
         :query            => (@auth ? { :auth => @auth }.merge(query) : query),
         :follow_redirect  => true
       })
     end
+
+    def non_relative_path_against_child_base_url?(path)
+      path.lstrip[0] == '/' && URI(@request.base_url).path != "/"
```

Comment:
Why is lstrip required? Paths should not contain leading whitespace. If they do, we should let the code explode normally, I believe.

The path check is also brittle: `URI('https://asdf.io').path` yields empty string. I would prefer to just always throw an error if path starts with a slash.